tests/test_memory_templates.py

```
@@ -0,0 +1,658 @@
+    """Verify that all expected port connections exist in instantiation.
+
+    Args:
+        instantiation_block: Port list string from module instantiation
+        expected_ports: Dict of {port_name: expected_connection_signal}
+
+    Returns:
+        Dict with verification results
+    """
+    results = {"found": {}, "missing": []}
+
+    for port_name, expected_signal in expected_ports.items():
```

Ruff (0.15.6)

[warning] Loop control variable `expected_signal` not used within loop body

Rename unused `expected_signal` to `_expected_signal`

(B007)